### PR TITLE
2480 - init public schema via ddl migration step

### PR DESCRIPTION
**Pull request by sorja**
Introduce new command: ```yarn migration-public:run```

Running the command the first time initialises the database as seen in the screenshots:

image
image

Running the command second time does nothing

### CODE REVIEW

File: src/server/repository/public/ddl/getCreatePublicSchemaDDL.ts [hunk 1]
```diff
@@ -0,0 +1,162 @@
+export const getCreatePublicSchemaDDL = (schemaName = 'public'): string => {
```
Comment (sorja) [inline src/server/repository/public/ddl/getCreatePublicSchemaDDL.ts:1]: Most of this is autogenerated

Some changes to generated file:

Added `if not exists` clases for enums, tables and extensions
Use bigserial for id columns
Added some comments (eg column `file` of `file` table)

File: src/server/repository/public/ddl/getCreatePublicSchemaDDL.ts [hunk 2]
```diff
@@ -0,0 +1,162 @@
...
+      name character varying default '',
+      published boolean not null default false,
+      foreign key (assessment_id) references ${schemaName}.assessment (id)
+        on update cascade on delete cascade
+    );
+    create unique index if not exists assessment_cycle_uuid_key on ${schemaName}.assessment_cycle using btree (uuid);
+
+    create table if not exists ${schemaName}.country (
+      country_iso character varying(3) primary key not null,
+      config jsonb
```
Comment (minotogna) [inline src/server/repository/public/ddl/getCreatePublicSchemaDDL.ts]: Should we have not null and default ?

File: src/server/repository/public/ddl/getCreatePublicSchemaDDL.ts [hunk 3]
```diff
@@ -0,0 +1,162 @@
...
+    );
+
+    create table if not exists ${schemaName}.file (
+      id bigserial primary key,
+      uuid uuid not null default uuid_generate_v4(),
+      name character varying(255) not null,
+      file bytea not null, -- TODO: Remove this when S3 branch merged
```
Comment (minotogna) [inline src/server/repository/public/ddl/getCreatePublicSchemaDDL.ts:55]: The s3 branch should have a migration to update this table (after this main branch is merged). Please add a todo in that PR

Comment (sorja) [reply src/server/repository/public/ddl/getCreatePublicSchemaDDL.ts:55]: Added

File: src/server/repository/public/ddl/getCreatePublicSchemaDDL.ts (hunk 2 above)
Comment (sorja) [reply src/server/repository/public/ddl/getCreatePublicSchemaDDL.ts]: Added

**Review by minotogna — APPROVED** [review]

**Pull request merged**